samples/bluetooth/channel_sounding_ras_initiator/src/main.c

```diff
@@ -0,0 +1,569 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @brief Channel Sounding Initiator with Ranging Requestor sample
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/bluetooth/cs.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/conn.h>
+#include <bluetooth/scan.h>
+#include <bluetooth/services/ras.h>
+#include <bluetooth/gatt_dm.h>
+#include "distance_estimation.h"
+
+#include <dk_buttons_and_leds.h>
+
+#define CON_STATUS_LED DK_LED1
+
+#define CS_CONFIG_ID	       0
+#define NUM_MODE_0_STEPS       1
+#define PROCEDURE_COUNTER_NONE (-1)
+
+static K_SEM_DEFINE(sem_remote_capabilities_obtained, 0, 1);
+static K_SEM_DEFINE(sem_config_created, 0, 1);
+static K_SEM_DEFINE(sem_cs_security_enabled, 0, 1);
+static K_SEM_DEFINE(sem_procedure_done, 0, 1);
+static K_SEM_DEFINE(sem_connected, 0, 1);
+static K_SEM_DEFINE(sem_rd_ready, 0, 1);
+static K_SEM_DEFINE(sem_discovery_done, 0, 1);
+static K_SEM_DEFINE(sem_mtu_exchange_done, 0, 1);
+static K_SEM_DEFINE(sem_rd_complete, 0, 1);
+static K_SEM_DEFINE(sem_security, 0, 1);
+
+static struct bt_conn *connection;
+static uint8_t n_ap;
+static uint8_t latest_num_steps_reported;
+NET_BUF_SIMPLE_DEFINE_STATIC(latest_local_steps, BT_RAS_PROCEDURE_MEM);
```

Comment:
I don't think BT_RAS_PROCEDURE_MEM is the correct estimate here, since the local steps are storing a full `bt_le_cs_subevent_step` and no subevent headers